pkg/apis/pipeline/v1alpha1/resource_types.go

```diff
@@ -123,6 +123,7 @@ type PipelineResourceResult struct {
 	Key         string              `json:"key"`
 	Value       string              `json:"value"`
 	ResourceRef PipelineResourceRef `json:"resourceRef,omitempty"`
+	ResultType  string              `json:"type,omitempty"`
```

Comment:
Im not super familiar with go lang, but perhaps something like and enum type is useful here. 

I think in go you can do something like this: 

```
	ResourceRef PipelineResourceRef `json:"resourceRef,omitempty"`
	ResultType  ResultType          `json:"type,omitempty"`
}

type ResultType string

const (
	PipelineResourceResultType = "PipelineResourceResult"
	TaskRunResultType          = "TaskRunResult"
	UnknownResultType          = ""
)
```

That way we don't have to hard code the strings everywhere and can use the constants instead.

Reply:
@sbwsg any thoughts?

Reply:
It seems reasonable to use an enum type so that we don't hard code the strings.  I don't think we necessarily need the `UnknownResultType` though since the contract here is entirely internal to tekton.  Technically any string that isn't "PipelineResourceResult" or "TaskRunResult" is unknown.  And if we saw a result type that we didn't know about then I think it'd just be an error.